bfd/peXXigen.c

```diff
@@ -4573,6 +4575,43 @@ _bfd_XXi_final_link_postscript (bfd * abfd, struct coff_final_link_info *pfinfo)
 #endif
     }
 
+  h1 = coff_link_hash_lookup (coff_hash_table (info),
+			      (bfd_get_symbol_leading_char (abfd) != 0
+			       ? "__load_config_used" : "_load_config_used"),
+			      false, false, true);
+  if (h1 != NULL)
+    {
+      char sz[4];
+      if ((h1->root.type == bfd_link_hash_defined
+	   || h1->root.type == bfd_link_hash_defweak)
+	  && h1->root.u.def.section != NULL
+	  && h1->root.u.def.section->output_section != NULL)
+	pe_data (abfd)->pe_opthdr.DataDirectory[PE_LOAD_CONFIG_TABLE].VirtualAddress =
+	  (h1->root.u.def.value
+	   + h1->root.u.def.section->output_section->vma
+	   + h1->root.u.def.section->output_offset
+	   - pe_data (abfd)->pe_opthdr.ImageBase);
+      else
+	{
+	  _bfd_error_handler
+	    (_("%pB: unable to fill in DataDictionary[10] because __load_config_used is missing"),
+	     abfd);
+	  result = false;
+	}
+      /* the size is stored as the first 4 bytes at _load_config_used */
+      if (bfd_get_section_contents (abfd, h1->root.u.def.section->output_section,
```

Comment:
I think it's possible to get these bytes directly form the input section. Looking at definition of `h1->root.u.def` (https://github.com/jeremyd2019/binutils-gdb/blob/master/include/bfdlink.h#L168-L173), I suspect `value` is the offset in that section, so this can be just 

```c
      if (h1->root.u.def.section != NULL
          && bfd_get_section_contents (abfd, h1->root.u.def.section, sz,
                                       h1->root.u.def.value, 4))
```

Reply:
> ```c
>       if (h1->root.u.def.section != NULL
>           && bfd_get_section_contents (abfd, h1->root.u.def.section, sz,
>                                        h1->root.u.def.value, 4))
> ```

This is not correct, because `h1->root.u.def.section` doesn't belong in `abfd`. In GDB if I `p abfd->filename` it gives the output EXE filename, so it's correct to use `h1->root.u.def.section->output_section`.

Issue here is that there should be proper checks for null pointers, and the `offset` value is incorrect.